### PR TITLE
Convert photo player to class

### PR DESCRIPTION
**Pull request by dkanada**
Here's an extremely small player we can use as a base to convert the rest of them.

### CODE REVIEW

**Comment by dkanada** [comment]
This should be merged in tandem with #1263 for the initialization change.

File: src/components/photoPlayer/plugin.js [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
+        this.name = 'Photo Player';
+        this.type = 'mediaplayer';
+        this.id = 'photoplayer';
+        this.priority = 1;
     }
 
-    PhotoPlayer.prototype.play = function (options) {
+    play(options) {
 
         return new Promise(function (resolve, reject) {
```
Comment (MrTimscampi) [inline src/components/photoPlayer/plugin.js:21]: We can replace the require() on 23 with import(), the Babel transform will convert it. Then you can remove the "require" dependency.

File: src/components/photoPlayer/plugin.js [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
-
-        var self = this;
-
-        self.name = 'Photo Player';
-        self.type = 'mediaplayer';
-        self.id = 'photoplayer';
-
-        // Let any players created by plugins take priority
-        self.priority = 1;
+import browser from 'browser';
```
Comment (MrTimscampi) [inline src/components/photoPlayer/plugin.js:1]: Unused dependency.

File: src/components/photoPlayer/plugin.js [hunk 3]
```diff
@@ -1,19 +1,22 @@
...
+import require from 'require';
+import events from 'events';
```
Comment (MrTimscampi) [inline src/components/photoPlayer/plugin.js:3]: Unused dependency.

File: src/components/photoPlayer/plugin.js [hunk 4]
```diff
@@ -1,19 +1,22 @@
...
+import appHost from 'apphost';
```
Comment (MrTimscampi) [inline src/components/photoPlayer/plugin.js:4]: Unused dependency.

File: src/components/photoPlayer/plugin.js [hunk 5]
```diff
@@ -1,19 +1,22 @@
...
+import loading from 'loading';
```
Comment (MrTimscampi) [inline src/components/photoPlayer/plugin.js:5]: Unused dependency.

File: src/components/photoPlayer/plugin.js [hunk 6]
```diff
@@ -1,19 +1,22 @@
...
+import dom from 'dom';
```
Comment (MrTimscampi) [inline src/components/photoPlayer/plugin.js:6]: Unused dependency.

File: src/components/photoPlayer/plugin.js [hunk 7]
```diff
@@ -1,19 +1,22 @@
...
+import playbackManager from 'playbackManager';
```
Comment (MrTimscampi) [inline src/components/photoPlayer/plugin.js:7]: Unused dependency.

File: src/components/photoPlayer/plugin.js [hunk 8]
```diff
@@ -1,19 +1,22 @@
...
+import appRouter from 'appRouter';
```
Comment (MrTimscampi) [inline src/components/photoPlayer/plugin.js:8]: Unused dependency.

**Comment by MrTimscampi** [comment]
@dkanada This needs the plugin loader change (to handle Promises)

File: src/components/photoPlayer/plugin.js [hunk 9]
```diff
@@ -1,23 +1,18 @@
...
+        this.priority = 1;
     }
 
-    PhotoPlayer.prototype.play = function (options) {
+    play(options) {
 
         return new Promise(function (resolve, reject) {
 
-            require(['slideshow'], function (slideshow) {
+            import('slideshow').then(slideshow => {
```
Comment (dmitrylyzo) [inline src/components/photoPlayer/plugin.js:15]: _Same as in #1089_
`TypeError: slideshow is not a constructor`
`slideshow = {default: ƒ}`

```suggestion
            import('slideshow').then(({default: slideshow}) => {
```

File: src/components/photoPlayer/plugin.js [hunk 10]
```diff
@@ -1,23 +1,18 @@
...
-
-        self.name = 'Photo Player';
-        self.type = 'mediaplayer';
-        self.id = 'photoplayer';
-
-        // Let any players created by plugins take priority
-        self.priority = 1;
+import connectionManager from 'connectionManager';
+
+export class PhotoPlayer {
```
Comment (thornbill) [inline src/components/photoPlayer/plugin.js:3]: Should we specify this as the `default` export?

File: src/components/photoPlayer/plugin.js (hunk 9 above)
Comment (dmitrylyzo) [reply src/components/photoPlayer/plugin.js:15]: Or here is an alternative suggestion https://github.com/jellyfin/jellyfin-web/pull/1089#discussion_r432112537
_if possible_

File: src/components/photoPlayer/plugin.js (hunk 10 above)
Comment (dkanada) [reply src/components/photoPlayer/plugin.js:3]: I used the same format we seem to be using everywhere else, which is adding it to the bottom of the file.

Comment (thornbill) [reply src/components/photoPlayer/plugin.js:3]: That seems odd. Why not just do `export default class PhotoPlayer`?

Comment (dkanada) [reply src/components/photoPlayer/plugin.js:3]: @thornbill I can add that in a separate pull request if that's the only other reservation you had about the changes here.

Comment (dkanada) [reply src/components/photoPlayer/plugin.js:3]: I just checked and this is actually the only instance, just removed it now.

Comment (dmitrylyzo) [reply src/components/photoPlayer/plugin.js]: > I just checked and this is actually the only instance, just removed it now.

You are not the only one who has exported `default` at the end of file: #1089 (`backdropScreensaver`)

Comment (dkanada) [reply src/components/photoPlayer/plugin.js]: Ah, well we haven't merged that yet so there's no problem quite yet.

**Pull request merged**